### PR TITLE
Azure: uses token and update doc and requirements

### PR DESCRIPTION
**Pull request by xp-1000**
- udpate readme with terraform-docs and fix examples
- requirement for azuread provider to 1.1: https://github.com/hashicorp/terraform-provider-azuread/issues/314
- fix depreacted "name" for azuread_application
- add signalfx token

### CODE REVIEW

File: cloud/azure/README.md [hunk 1]
```diff
@@ -92,17 +116,20 @@ variable "azure_tenant_id" {
...
-  subscription_id = var.azure_subscription_id
-  tenant_id       = var.azure_tenant_id
+  tenant_id = var.azure_tenant_id
 }
 
 ```
 
 ## Notes
 
+* This module will create an organization token and use it for ingesting data from the created GCP integration.
```
Comment (jmapro) [inline cloud/azure/README.md:131]:
```suggestion
* This module will create an organization token and use it for ingesting data from the created Azure integration.
```

File: cloud/azure/sfx/README.md [hunk 2]
```diff
@@ -68,5 +87,7 @@ provider "signalfx" {
 
 ## Notes
 
+* This module will create an organization token and use it for ingesting data from the created GCP integration.
```
Comment (jmapro) [inline cloud/azure/sfx/README.md:90]:
```suggestion
* This module will create an organization token and use it for ingesting data from the created Azure integration.
```

File: cloud/azure/sfx/integrations-azure.tf [hunk 3]
```diff
@@ -17,3 +17,24 @@ resource "signalfx_azure_integration" "azure_integration" {
...
+
+# "Named token to use for ingest on the SignalFx Azure integration"
+resource "signalfx_org_token" "azure_integration" {
+  name        = local.integration_name
+  description = "Org token for ingesting data from ${local.integration_name} Azure integration"
+
+  # Where to send notifications about this token's limits. Please consult the Notification Format laid out in detectors
+  # notifications = ["Email,foo-alerts@bar.com"]
+
+  # host_or_usage_limits {
```
Comment (jmapro) [inline cloud/azure/sfx/integrations-azure.tf:29]: Can it be a variable to allow to override default settings ?

File: cloud/azure/README.md (hunk 1 above)
Comment (xp-1000) [reply cloud/azure/README.md:131]: good catch thanks

File: cloud/azure/sfx/integrations-azure.tf (hunk 3 above)
Comment (xp-1000) [reply cloud/azure/sfx/integrations-azure.tf:29]: yes it is planned !

but I would like to merge this one and do that in another PR for all cloud integrations (including gcp and aws)

**Review by Shr3ps — APPROVED** [review]

**Review by jmapro — APPROVED** [review]

**Pull request merged**